cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java

```diff
@@ -67,8 +67,24 @@ protected AbstractStorageProviderNamespaceAdmin(CConfiguration cConf,
   @Override
   public void create(NamespaceMeta namespaceMeta) throws IOException, ExploreException, SQLException {
     Location namespaceHome;
-    String rootDirectory = namespaceMeta.getConfig().getRootDirectory();
-    if (Strings.isNullOrEmpty(rootDirectory)) {
+    if (hasCustomLocation(namespaceMeta)) {
```

Comment:
@rohitsinha54 lets move the filesystem stuff to its own create/delete methods. this code is going to get unmanageable soon, after HBase and Hive stuff is added